### PR TITLE
Adding npe guarding

### PR DESCRIPTION
**Pull request by marcingrzejszczak**

### CODE REVIEW

File: accurest-core/src/main/groovy/io/coderate/accurest/builder/SpockMethodBodyBuilder.groovy [hunk 1]
```diff
@@ -18,8 +18,10 @@ class SpockMethodBodyBuilder {
 		blockBuilder.addLine('given:').startBlock()
 		blockBuilder.addLine('def request = given()')
 		blockBuilder.indent()
-		stubDefinition.request.headers.headers.collectEntries { [(it.name): it.serverValue] }.each { Map.Entry entry ->
-			blockBuilder.addLine(".header('${entry.key}', '${entry.value}')")
+		if (stubDefinition.request.headers) {
+			stubDefinition.request.headers.headers.collectEntries { [(it.name): it.serverValue] }.each { Map.Entry entry ->
```
Comment (szpak) [inline accurest-core/src/main/groovy/io/coderate/accurest/builder/SpockMethodBodyBuilder.groovy]: Haven't tested, but maybe `stubDefinition.request.headers?.headers.collectEntries` could help?

Comment (marcingrzejszczak) [inline accurest-core/src/main/groovy/io/coderate/accurest/builder/SpockMethodBodyBuilder.groovy]: you would have to add more ? that's why I'd prefer to add expiclit check with the if. IMO it's more readable

**Comment by szpak** [comment]
+1

**Comment by jkubrynski** [comment]
I've refactored this part in 736ae1b79063de3840d852e72f21a5c1f31c1a89

**Pull request closed without merging**